src/transport/WsOutboundTransporter.ts

```diff
@@ -1,12 +1,21 @@
 import { OutboundTransporter } from './OutboundTransporter'
 import { Agent } from '../agent/Agent'
-import { WebSocketTransport } from '../agent/TransportService'
+import { TransportSession } from '../agent/TransportService'
 import { Logger } from '../logger'
 import { ConnectionRecord } from '../modules/connections'
 import { OutboundPackage } from '../types'
 import { Symbols } from '../symbols'
 import { WebSocket } from '../utils/ws'
 
+export class WebSocketTransportSession implements TransportSession {
+  public readonly type = 'websocket'
+  public socket?: WebSocket
+
+  public constructor(socket?: WebSocket) {
+    this.socket = socket
+  }
+}
+
```

Comment:
Is the transport session for the inbound transport, outbound transport or both?

Reply:
Hmm. That's a good question. I would say it's for both. A developer sets the session when calling `agent.recieveMessage(message, session)` passing inbound message. Then it can be used for outbound messages.